File: my-folder/0392-is-subsequence/solution.c
bool isSubsequence(char* s, char* t) {
    bool result = true;

    size_t i = 0, j = 0;
    while (s[i]) {
        while (t[j] && (t[j] != s[i]))
            j++;
        if (!t[j]){
            result = false;
            break;
        }
        i++;j++;
    }

    return result;
}

File: my-folder/0392-is-subsequence/solution.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut s_chars = s.chars();
        let mut current_char = s_chars.next();

        for t_char in t.chars() {
            if let Some(c) = current_char {
                if c == t_char {
                    current_char = s_chars.next();
                }
            } else {
                break;
            }
        }

        current_char.is_none()
    }
}
